lib/flake-info.nix

```diff
@@ -0,0 +1,34 @@
+# This function produces a lib overlay to be used by the nixpkgs
+# & nixpkgs/lib flakes to provide meaningful values for
+# `lib.trivial.version` et al..
+#
+# Internal and subject to change, don't use this anywhere else!
+# Instead, consider using a public interface, such as this flake here
+# in this directory, `lib/`, or use the nixpkgs flake, which applies
+# this logic for you in its `lib` output attribute.
+
+self: # from the flake
+
+finalLib: prevLib: # lib overlay
+
+{
+  trivial = prevLib.trivial // {
+    versionSuffix =
+      ".${finalLib.substring 0 8 (self.lastModifiedDate or "19700101")}.${self.shortRev or "dirty"}";
+    revisionWithDefault = default: self.rev or default;
+
+    # this overrides lib/trivial.nix, in which isFlake is unconditionally set
+    # to false.
+    isFlake = true;
```

Comment:
Note that this may suffer from a similar expectation, but that expectation is less realistic to be executed, unless we give `lib` a special treatment in Nix's `call-flake.nix`. I don't like that idea.